mypy/subtypes.py

```diff
@@ -720,4 +720,4 @@ def is_more_precise(t: Type, s: Type) -> bool:
             # Fall back to subclass check and ignore other properties of the callable.
             return is_proper_subtype(t.fallback, s)
         return is_proper_subtype(t, s)
-    return sametypes.is_same_type(t, s)
+    return is_proper_subtype(t, s)
```

Comment:
Why do we need a special case when ``t`` is ``CallableType`` and ``s`` is ``Instance``? It looks like ``ProperSubtypeVisitor`` above does exactly the same. So that the whole function boils down to:
```python
    if isinstance(s, AnyType):
        return True
    return is_proper_subtype(t, s)
```